### ISSUE TITLE
3. Viewing of Videos

### ISSUE THREAD

**Issue by the-muses** [the-muses-ltd/Cassandra#10]
Ability to watch lecture video (linked to youtube and Vimeo, external link and our own from our channel).

Viewing Portal
Connected to Links and categories in backend.

**Comment by edward-hong** [comment]
I've tested out the `react-player` module. Playing external videos on Youtube and Video works great. 

Setting the `loop` prop to true makes the video start from the beginning once it ends. This works on Vimeo but doesn't seem to work on Youtube.

The `controls` prop is also a boolean that toggles native controls to the user. This works on Youtube but not on Vimeo however this is by design on Vimeo's end.

I've also tested the player with local video files. The local files are set relative to the path of the `public` folder where `index.html` currently is. Here are the formats that works from my findings:

- `.mp4` works
- `.ogv` works
- `.webm` works
- `.flv` doesn't work

Currently, I don't think a PR is needed as we don't have any actual videos that need to present on Cassandra. I'm posting this here just confirm that `react-player` is quite good and fit our needs when they do arise.
